docs/messaging/messaging-spans.md

```diff
@@ -196,21 +198,74 @@ Examples:
 * `AuthenticationRequest-Conversations process`
 * `(anonymous) publish` (`(anonymous)` being a stable identifier for an unnamed destination)
 
-### Span kind
-
-A producer of a message should set the span kind to `PRODUCER` unless it synchronously waits for a response: then it should use `CLIENT`.
-The processor of the message should set the kind to `CONSUMER`, unless it always sends back a reply that is directed to the producer of the message
-(as opposed to e.g., a queue on which the producer happens to listen): then it should use `SERVER`.
-
 ### Operation names
 
 The following operations related to messages are defined for these semantic conventions:
 
 | Operation name | Description |
 | -------------- | ----------- |
-| `publish`      | A message is sent to a destination by a message producer/client.       |
-| `receive`      | A message is received from a destination by a message consumer/server. |
-| `process`      | A message that was previously received from a destination is processed by a message consumer/server. |
+| `publish`      | One or more messages are provided for publishing to an intermediary. If a single message is published, the context of the "Publish" span can be used as the creation context and no "Create" span needs to be created. |
+| `create`       | A message is created. "Create" spans always refer to a single message and are used to provide a unique creation context for messages in batch publishing scenarios. |
+| `receive`      | One or more messages are requested by a consumer. This operation refers to pull-based scenarios, where consumers explicitly call methods of messaging SDKs to receive messages. |
+| `deliver`      | One or more messages are passed to a consumer. This operation refers to push-based scenarios, where consumer register callbacks which get called by messaging SDKs. |
+
+### Span kind
+
+[Span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)
+SHOULD be set according to the following table, based on the operation a span describes.
+
+| Operation name | Span kind|
+|----------------|-------------|
+| `publish`      | `PRODUCER`, if the context of the "Publish" span is used as creation context. |
+| `create`       | `PRODUCER`, |
+| `receive`      | `CONSUMER` |
+| `deliver`      | `CONSUMER` |
+
+For cases not covered by the table above, the span kind should be set according
+to the [generic specification about span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind),
+e. g. it should be set to CLIENT for the "Publish" span if its context is not
+used as creation context and if the "Publish" span models a synchronous call to
+the intermediary.
+
+Setting span kinds according to this table ensures that span links between
```

Comment:
Does "span links" here does refer to actual links or more of a "connection" between them? If it's the latter, I think we may need to remove the word "link" here to avoid confusion.

Reply:
I'm inclined to leave the reference to "links" in place, as it's indeed a link that we guarantee between producer and consumer spans. There might or might not be a parent-child relationship in addition (we're very opaque about that now as we removed any reference to it), but there will always be a link.